test/unit/test_exports/test_trap_density_xdmf_export.py

```diff
@@ -129,12 +130,16 @@ def test_trap_density_xdmf_export_traps_materials_mixed(tmpdir):
     my_model.initialise()
     my_model.run()
 
-    V = FunctionSpace(my_model.mesh.mesh, "CG", 1)
-
-    density_expected = interpolate(festim.as_expression(density_expr), V)
+    V = FunctionSpace(my_model.mesh.mesh, "DG", 1)
+    volume_markers = my_model.mesh.volume_markers
+    dx = Measure("dx", domain=my_model.mesh.mesh, subdomain_data=volume_markers)
+    density_expected = Function(V)
+    v = TestFunction(V)
+    F = inner(density_expected, v) * dx
+    F -= inner(festim.as_expression(density_expr), v) * dx(1)
```

Comment:
no need for `as_expression` here
```suggestion
    F -= inner(density_expr, v) * dx(1)
```